[Fix] Improve dim_size handling in SetTransformerAggregation to prevent CUDA crash

KAVYANSHTYAGI opened the pull request:
This PR improves the robustness of `SetTransformerAggregation` by:

- Automatically setting `dim_size = index.max() + 1` if `dim_size` is not provided.
- Raising a clear error if `index.max() >= dim_size` to avoid CUDA crashes during evaluation.

This is helpful especially for datasets like PPI where `data.batch` may be missing. It replaces hard-to-debug GPU errors with clear and early validation.

KAVYANSHTYAGI committed:
Update set_transformer.py

[Fix] Add dim_size validation and fallback to SetTransformerAggregation

KAVYANSHTYAGI committed:
Merge pull request #1 from KAVYANSHTYAGI/KAVYANSHTYAGI-patch-1

base repository: pyg-team/pytorch_geometric ← compare: KAVYANSHTYAGI:fix/set-transformer-aggregation-index-check

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rusty1s commented on torch_geometric/nn/aggr/set_transformer.py:
```diff
@@ -94,6 +94,15 @@ def forward(
         max_num_elements: Optional[int] = None,
     ) -> Tensor:
 
+        if dim_size is None:
```
I think this is already handled in `to_dense_batch`.

rusty1s commented on torch_geometric/nn/aggr/set_transformer.py:
```diff
@@ -94,6 +94,15 @@ def forward(
+        if dim_size is None:
+            dim_size = int(index.max()) + 1
+
+        if int(index.max()) >= dim_size:
+            raise ValueError(
+                f"SetTransformerAggregation error: index.max() = {int(index.max())}, "
+                f"but dim_size = {dim_size}. This causes an indexing error on GPU. "
+                f"Ensure data.batch is set or dim_size is passed explicitly.")
```
This leads to a device sync, so we should avoid this.

KAVYANSHTYAGI committed:
Merge branch 'pyg-team:master' into fix/set-transformer-aggregation-dim-check-v2

KAVYANSHTYAGI committed:
Update set_transformer.py

Fix: Remove device sync and dim_size fallback in SetTransformerAggregation

- Removed redundant dim_size = index.max() + 1 logic (handled in to_dense_batch).
- Added GPU-safe index validation to avoid CUDA crashes.

KAVYANSHTYAGI reviewed (commented):
Thank you for the feedback!

I've removed the fallback dim_size = index.max() + 1 to avoid redundancy with to_dense_batch, as suggested.

Also eliminated device sync by replacing the .max() check with a GPU-safe tensor comparison using (index >= dim_size).any().

Let me know if any further simplification is preferred!

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
